engine/api/pipeline/audit.go

```diff
@@ -109,3 +131,38 @@ func (p *PipelineAudit) PostInsert(s gorp.SqlExecutor) error {
 	}
 	return nil
 }
+
+const keepAudits = 50
+
+func purgeAudits(db gorp.SqlExecutor) error {
+	var nbAuditsPerPipelinewID = []struct {
+		PipelineD int64 `db:"pipeline_id"`
+		NbAudits  int64 `db:"nb_audits"`
+	}{}
+
+	query := `select pipeline_id, count(id) "nb_audits" from pipeline_audit group by pipeline_id having count(id)  > $1`
+	if _, err := db.Select(&nbAuditsPerPipelinewID, query, keepAudits); err != nil {
+		return sdk.WithStack(err)
+	}
+
+	for _, r := range nbAuditsPerPipelinewID {
+		log.Debug("purgeAudits> deleting audits for pipeline %d (%d audits)", r.PipelineD, r.NbAudits)
+		var ids []int64
+		query = `select id from pipeline_audit where pipeline_id = $1 order by versionned asc offset $2`
+		if _, err := db.Select(&ids, query, r.PipelineD, keepAudits); err != nil {
+			return sdk.WithStack(err)
+		}
+		for _, id := range ids {
+			if err := deleteAudit(db, id); err != nil {
+				log.Error("purgeAudits> unable to delete audit %d: %v", id, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func deleteAudit(db gorp.SqlExecutor, id int64) error {
+	_, err := db.Exec(`delete from pipeline_audit where id = $1`, id)
```

Comment:
where id = ANY... we can remove all the audit for each pipeline in one time to reduce request count.

Reply:
I think it's to avoid statement timeout if there is too much audits

Reply:
yep